### PR TITLE
new function removeUmlaut

### PR DESCRIPTION
**Pull request by weirichs**
see the corresponding mail ... I'm not sure whether the function is appropriate for it's purposes ...

### CODE REVIEW

**Comment by beckerbenj** [comment]
Note:
R CMD check currently fails because of factor level ordering. It is a bit tricky to debug, as I can not reproduce this result outside of R CMD check. Will tackle this somewhen later this week, if that is ok with everyone.

**Comment by grewered** [comment]
@beckerbenj I haven't found anything regarding best practices and assertions/input validation for s3. I didn't find the source code for `na.omit` either. For now I would leave the assertions in the general function then.

**Comment by beckerbenj** [comment]
Thanks for looking into this @grewered. Looks good to me 👍 

@weirichs :
Have fixed the problematic tests; could you review everything? (can't formally request a review since you opened the PR)

File: R/cleanifyString.r [hunk 1]
```diff
@@ -1,8 +1,18 @@
...
   UseMethod("cleanifyString")
   }
 
-cleanifyString.character <- function(x, removeNonAlphaNum = TRUE, replaceSpecialChars = TRUE, oldEncoding = NULL, ...) {
+### ----------------------------------------------------------------------------
+
+cleanifyString.character <- function(x, removeNonAlphaNum = TRUE, replaceSpecialChars = TRUE,
+                                     oldEncoding = NULL, ...) {
+  lapply(c(removeNonAlphaNum, replaceSpecialChars, oldEncoding),
+         checkmate::assert_logical, len = 1)
```
Comment (weirichs) [inline R/cleanifyString.r]: The first assertion (according to logical) is fine. I think the second consistency check (`assert_character(x)`) could be omitted. Benjamin has changed the function towards S3 methods. This means that `cleanifyString.character()` can only be called if the class has already been confirmed to be character.

File: R/cleanifyString.r [hunk 2]
```diff
@@ -15,22 +25,36 @@ cleanifyString.character <- function(x, removeNonAlphaNum = TRUE, replaceSpecial
...
   return(varNew)}
 
-cleanifyString.data.frame <- function(x, removeNonAlphaNum = TRUE, replaceSpecialChars = TRUE, oldEncoding = NULL, ...) {
+### ----------------------------------------------------------------------------
+
+cleanifyString.data.frame <- function(x, removeNonAlphaNum = TRUE, replaceSpecialChars = TRUE,
+                                      oldEncoding = NULL, ...) {
+  lapply(c(removeNonAlphaNum, replaceSpecialChars, oldEncoding),
+         checkmate::assert_logical, len = 1)
+  x <- makeDataFrame(x)
```
Comment (weirichs) [inline R/cleanifyString.r]: I think, `x <- makeDataFrame(x)` should be omitted. If `x` were something else (a tibble, for example), the S3 method `cleanifyString.data.frame` would not even be called.

**Review by weirichs — COMMENTED** [review]
Thanks! I was unsure about some points, so I would like to consult Benjamin again if necessary.

**Comment by weirichs** [comment]
> Thanks for looking into this @grewered. Looks good to me 👍
> 
> @weirichs : Have fixed the problematic tests; could you review everything? (can't formally request a review since you opened the PR)

Thanks @grewered and @beckerbenj ... I've reviewed the changes and merged them into master!

**Pull request merged**